### ISSUE TITLE
Possible inconsistency in reported `mae/depa` train/test metrics

### ISSUE THREAD

**Issue by hprats** [ICAMS/grace-tensorpotential#30]
Hi,

I am training several GRACE models and noticed a surprising trend in the final reported metrics. For all the hyperparameter variants I tested, the reported `mae/depa` is consistently higher for the training set than for the test set. However, the other metrics behave as expected:

- `mae/depa`: train > test
- `rmse/depa`: train < test
- `mae/f_comp`: train < test
- `rmse/f_comp`: train < test

Since only `mae/depa` shows this behavior, I was wondering whether this could be a reporting issue, for example whether the train/test values might be printed or written in the wrong order for this specific metric, or whether `mae/depa` is computed in a way that could explain this.

Below is one example input.yaml file:

```
seed: 8
cutoff: 6.0

data:
  filename: dataset_g3-40.pckl.gzip
  test_size: 0.2
  reference_energy: 0

potential:
   # If elements not provided - determined automatically from data
  preset: GRACE_2LAYER # LINEAR, FS, GRACE_1LAYER, GRACE_2LAYER
  kwargs: {lmax: [3, 3], max_order: 4, n_rad_max: [32, 48], n_mlp_dens: 10}

  scale: True # False/True or float
fit:
  loss: {
    energy: { weight: 1, type: huber , delta: 0.01 },
    forces: { weight: 5, type: huber , delta: 0.01 },

    switch: { after_iter: 600, learning_rate: 0.001, energy: { weight: 5 }, forces: { weight: 2 }, }
  }

  maxiter: 1200 # Number of epochs / iterations
  optimizer: Adam
  opt_params: {
            learning_rate: 0.01,
            amsgrad: True,
            use_ema: True,
            ema_momentum: 0.99,
            weight_decay: null,
            clipvalue: 1.0,
        }

  learning_rate_reduction: { patience: 5, factor: 0.98, min: 5.0e-4, stop_at_min: True, resume_lr: True, }

  compute_convex_hull: False
  batch_size: 4 # Important hyperparameter for Adam and irrelevant (but must be) for L-BFGS-B
  test_batch_size: 128 # test batch size (optional)

  jit_compile: True
  eval_init_stats: False # to evaluate initial metrics

  train_max_n_buckets: 10 # max number of buckets (group of batches of same shape) in train set
  test_max_n_buckets: 5 # same for test

  checkpoint_freq: 2 # frequency for **REGULAR** checkpoints.
  # save_all_regular_checkpoints: True # to store ALL regular checkpoints
  progressbar: True # show batch-evaluation progress bar
  train_shuffle: True # shuffle train batches on every epoch
```

And those are the final reported metrics:

> 2026/05/08 10:06:17 I - Iteration #827/1200 TRAIN(TEST): total_loss: 1.271e-03 (1.983e-03) mae/depa: 2.344e-03 (2.256e-03) rmse/depa: 3.113e-03 (4.825e-03) mae/f_comp: 7.108e-02 (1.029e-01) rmse/f_comp: 1.050e-01 (1.770e-01) Time(mcs/at): 110 (19)

**Comment by yury-lysogorskiy** [comment]
You can take final model(note that this is usually best test loss model) and load it as ASE TPCalculator and compute these metrics manually. That would be the first thing I would try

**Comment by hprats** [comment]
Thanks! I've recomputed the metrics and now they show the expected behaviour (train < test).
